pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SegmentPushUtils.java

```diff
@@ -458,10 +438,71 @@ public static void sendSegmentsUriAndMetadata(SegmentGenerationJobSpec spec, Pin
         });
       }
     } finally {
-      for (Map.Entry<String, File> metadataFileEntry: segmentMetadataFileMap.entrySet()) {
+      for (Map.Entry<String, File> metadataFileEntry : segmentMetadataFileMap.entrySet()) {
         FileUtils.deleteQuietly(metadataFileEntry.getValue());
       }
-      FileUtils.forceDelete(allSegmentsMetadataTarFile);
+      if (allSegmentsMetadataTarFile != null) {
+        FileUtils.deleteQuietly(allSegmentsMetadataTarFile);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  public static void generateSegmentMetadataFiles(SegmentGenerationJobSpec spec, PinotFS fileSystem,
```

Comment:
Can we make the following changes:
1) Reduce visibility for this method to package level
2) Pass in the executor service as an arg to this method
3) Close the executor service in the finally block of the calling method.

Reply:
Done.